src/main/java/org/jenkinsci/plugins/mesos/MesosRetentionStrategy.java

```diff
@@ -86,7 +86,11 @@ private long checkInternal(MesosComputer c) {
 
       if (idleMilliseconds > MINUTES.toMillis(idleTerminationMinutes)) {
         LOGGER.info("Disconnecting idle computer " + c.getName());
-        c.getNode().terminate();
+        c.getNode().setPendingDelete(true);
```

Comment:
Because then you end up terminating a slave that's still doing things in the post build state, leading to a failed build. Rather we should wait until idle time has elapsed

On June 21, 2015 3:21:28 PM CDT, Vinod Kone <notifications@github.com> wrote:
>> @@ -86,7 +86,11 @@ private long checkInternal(MesosComputer c) {
>>  
>>        if (idleMilliseconds >
>MINUTES.toMillis(idleTerminationMinutes)) {
>>          LOGGER.info("Disconnecting idle computer " + c.getName());
>> -        c.getNode().terminate();
>> +        c.getNode().setPendingDelete(true);
>
>Why the change here to mark it for deletion instead of directly
>terminate it?
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/jenkinsci/mesos-plugin/pull/121/files#r32897165

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.